control-plane-operator/controllers/hostedcontrolplane/kas/params.go

```diff
@@ -101,7 +101,7 @@ func NewKubeAPIServerParams(ctx context.Context, hcp *hyperv1.HostedControlPlane
 	case hyperv1.Unmanaged:
 		params.EtcdURL = hcp.Spec.Etcd.Unmanaged.Endpoint
 	case hyperv1.Managed:
-		params.EtcdURL = config.DefaultEtcdURL
+		params.EtcdURL = fmt.Sprintf("https://etcd-client.%s.svc:2379", hcp.Namespace)
```

Comment:
Nit: Appending the full domain and using a trailing dot saves a couple of dns lookups as it won't iterate through all of the search domains